// xvideosx/lib/api/videos/details/parseResponse.js
const cheerio = require('cheerio');
const parseVideo = require('./parseVideo');
const parseRelated = require('./parseRelated');
const parseCategory = require('./parseCategory');
const parseModels = require('./parseModels');


const getVideo = ($) =>{
    return parseVideo($);
}
const getRelated = ($) => {
    return parseRelated($);
}
const getCategories = ($) =>{
    return parseCategory($);
}
const getModels = ($) =>{
    return parseModels($);
}
const parseResponse = ({ data }) => {
    const $ = cheerio.load(data);
    const video = getVideo($);
    const related = getRelated($);
    const categories = getCategories($);
    const models = getModels($);

    
    return{
        title: video.title,
        duration: video.duration,
        image: video.image,
        categories,
        channel: models.mainUploader,
        models: models.models,
        videoType: video.videoType,
        videoWidth: video.videoWidth,
        videoHeight: video.videoHeight,
        views: video.views,
        files: video.files,
        videos: related
    }
}

module.exports = parseResponse
